ssg/entities/common.py

```diff
@@ -309,3 +349,85 @@ def update_with(self, rhs):
         updated_refinements = self._subtract_refinements(extended_refinements)
         updated_refinements.update(self.refine_rules)
         self.refine_rules = updated_refinements
+
+
+class Templatable(object):
+
+    KEYS = dict(
+        template=lambda: None,
+    )
+
+    def __init__(self):
+        pass
+
+    def is_templated(self):
+        return isinstance(self.template, dict)
+
+    def get_template_context(self, env_yaml):
+        # TODO: The first two variables, 'rule_id' and 'rule_title' are expected by some
+        #       templates and macros even if they are not rendered in a rule context.
+        #       Better name for these variables are 'entity_id' and 'entity_title'.
+        return {
+            "rule_id": self.id_,
```

Comment:
You can add `entity_id` as well, so the transition period can start.

Reply:
I would rather just rename it in a separate self-contained PR, without any backward-compatibility. We have enough legacy lingering around.

Reply:
OK.